src/shipit_code_coverage/shipit_code_coverage/taskcluster.py

```diff
@@ -12,25 +12,36 @@
 queue_base = 'https://queue.taskcluster.net/v1/'
 
 
-def _get_build_platform_name(platform):
-    if platform == 'linux':
-        return 'linux64-ccov-opt'
-    elif platform == 'win':
-        return 'win64-ccov-debug'
-    else:
-        raise Exception('Unsupported platform: %s' % platform)
+class TaskclusterException(Exception):
+    pass
 
 
 def get_task(branch, revision, platform):
-    r = requests.get(index_base + 'task/gecko.v2.{}.revision.{}.firefox.{}'.format(branch, revision, _get_build_platform_name(platform)))
-    task = r.json()
-    if r.status_code == requests.codes.ok:
-        return task['taskId']
+    if platform == 'linux':
+        # A few days after https://bugzilla.mozilla.org/show_bug.cgi?id=1457393 is fixed,
+        # we should only have 'linux64-ccov-debug' here and only support one platform_name
+        # and no fallback.
+        platform_names = ['linux64-ccov-debug', 'linux64-ccov-opt']
+    elif platform == 'win':
+        platform_names = ['win64-ccov-debug']
     else:
-        if task['code'] == 'ResourceNotFound':
-            raise Exception('Code coverage build failed and was not indexed.')
-        else:
-            raise Exception('Unknown TaskCluster index error.')
+        raise TaskclusterException('Unsupported platform: %s' % platform)
+
+    while True:
```

Comment:
I would have used a for loop here to avoid the `while True`, saving the last exception raised in `except` and re-raising it outside the loop

Reply:
Good idea, I usually prefer plain `raise` rather than `raise EXC`, but in this case it shouldn't matter.

Reply:
Actually, I've used enumerate in the for loop now, so I can detect the last iteration and throw the exception if we're in the last iteration.